vss_invoice_mgmt/src/pages/business/pages/masters/vendormaster/pages/VendorMasterList.jsx

```diff
@@ -0,0 +1,205 @@
+import React, { useEffect, useState } from "react";
+import AdminNavbar from "../../../../../../components/Navbars/AdminNavbar";
+import Sidebar from "../../../../../../components/Sidebar/Sidebar";
+import HeaderStats from "../../../../../../components/Headers/HeaderStats";
+import FooterAdmin from "../../../../../../components/Footers/FooterAdmin";
+import { TextField } from "@mui/material";
+import { Button } from "@mui/material";
+import { NavLink, useNavigate } from "react-router-dom";
+import {
+  GetVendorMasterList,
+  PutVendorMasterList,
+} from "../vendormasterconfig.js";
+import Stack from "@mui/material/Stack";
+import CircularProgres from "@/components/Circular-Progress/CircularProgress";
+import Pagination from "@mui/material/Pagination";
+import StatusSelect from "@/components/Status-Select/Status_Select";
+
+const VendorMasterList = () => {
+  const [searchName, setSearchName] = useState("");
+  const [searchContactNo, setSearchContactNo] = useState("");
+  const [searchStatus, setSearchStatus] = useState("");
+  const [vendormasterlist, setVendorMasterList] = useState([]);
+  const [currentPage, setCurrentPage] = useState(1);
+  const [loading, setLoading] = useState(false);
+  const [totalpages, settotalpages] = useState(1);
+  const itemsPerPage = 10;
+  const navigate = useNavigate();
+
+  const tableHead = {
+    col: [
+      "Sr No",
+      "Vendor Name",
+      "Vendor Contact No.",
+      "Status",
+      "Edit",
+      "Block",
+    ],
+  };
+
+  const getVendorList = async (page, name, contactNo, status) => {
+    setLoading(true);
+    try {
+      const users = await GetVendorMasterList(
+        1,
+        page,
+        itemsPerPage,
+        name,
+        contactNo,
+        status
+      );
+      setVendorMasterList(users.data);
+      settotalpages(Math.ceil(users.totalCount / itemsPerPage));
+    } catch (error) {
+      setVendorMasterList([]);
+      settotalpages(1);
+    }
+    setLoading(false);
+  };
+
+  useEffect(() => {
```

Comment:
If we are using this then what is the purpose of search button?

Reply:
We search name by using search button only, if we cant use this then after search when we remove data into name field and our fields is empty so at bottom it didnt show our data back, by use this when our input fields empty it automatically show data at bottom.

Reply:
But this useEffect unnecessarily check values every time whenever use type. We don't this code.